### ISSUE TITLE
mixin constructor wrongly typed

### ISSUE THREAD

**Issue by npenin** [microsoft/TypeScript#39539]
**TypeScript Version:**  3.9.2

**Search Terms:** mixin constructor signature

**Code**

```ts
class A
{
    constructor(public x:number){}
}

function f<T extends { new(...args: any[]): object }>(cls:T) {
    return class DynamicProxy extends cls {
        constructor(...args: any[]) {
            super(1);
        }
    }
}

const B = f(A);
var b = new B();
b.x;```

**Expected behavior:** Either make it a warning or just allow any constructor signature.

**Actual behavior:** Errors with `Expected 1 arguments, but got 0.(2554)`

**Playground Link:** https://www.typescriptlang.org/play/index.html?ssl=1&ssc=1&pln=16&pc=5#code/MYGwhgzhAECCBQBve1XWAewHYQC4CcBXYXDfACgAdCAjEAS2GgA8AuLQgWxoFN8BKRAF94I+ADNCWEvWzRxAHgAq0Hs1w8sAExiJoWHgHdyAOjNh8AcwitoYLAE8A2gF1+tjDQBWPEtCEAfOSgNkr80Mho0Pg8uIT4WOjgUNAAIg5YYJyMAAr4GMwOquqaOkm6KFFomDgExKQUZiYW1rb2zm4RlVVVEISUfOQAjPwA3N1VIlEiYjV40ABC0AC88uSwY-AAbhbQNCv6RovkmzQmzKNAA

**Related Issues:** <!-- Did you find other bugs that looked similar? -->

**Comment by npenin** [comment]
The concept here is simplified to its maximum. My issue is that I am using dependency injection, and therefore, there is no parameter to be expected from this mixin
